js/ui/map.js

```diff
@@ -734,6 +734,20 @@ class Map extends Camera {
     }
 
     /**
+     * Returns a Boolean indicating whether the source is loaded.
+     *
+     * @param {string} id The ID of the source to be checked.
+     * @returns {boolean} A Boolean indicating whether the source is loaded.
+     */
+    isSourceLoaded(id) {
+        const source = this.style && this.style.sourceCaches[id];
+        if (source === undefined) {
+            throw new Error(`There is no source with ID '${id}'`);
```

Comment:
Per https://github.com/mapbox/mapbox-gl-js/issues/3208, we have a convention that we emit an `error` event rather than `throw` an error.